multipart.go

```diff
@@ -0,0 +1,65 @@
+package goreq
+
+import (
+	"bytes"
+	"io"
+	"mime/multipart"
+)
+
+// MultipartForm wraps functionality from the stdlib mime/multipart
+// You can supply any io.Reader as the source, and goreq will send your
+// request as a multipart form. For convenience, you can set the
+// "FileName" attribute and that will be added to the field header
+
+// Simple usage to upload a file looks like this
+//    file, _ := os.Open("/tmp/myfile")
+//    body := goreq.MultipartForm{
+//        Field:    "file",
+//        FileName: "myfile",
+//        Source:   file,
+//    }
+//    resp, _ := goreq.Request{
+//        Uri:          "http://some.place/uristuff",
+//        Method:       "POST",
+//        Body:         body,
+//        ContentType:  "multipart/mixed",
+//    }.Do()
+type MultipartForm struct {
+	Params   map[string]string
+	Field    string
+	Source   io.Reader
+	buffer   bytes.Buffer
+	FileName string
+}
+
+func (m *MultipartForm) ToReader() (_ io.Reader, err error) {
```

Comment:
@ryansb this function is not tested. I believe it's important to have a multipart_test.go that actually tests that `ToReader()` is doing what it's supposed to do.

Seems like test should validate that: 

- Extra params are populated correctly
- Filename field is used when supplied